schematicbrushreborn-core/src/main/java/de/eldoria/schematicbrush/config/sections/presets/DatabasePresetRegistry.java

```diff
@@ -0,0 +1,284 @@
+package de.eldoria.schematicbrush.config.sections.presets;
+
+import com.google.gson.Gson;
+import de.eldoria.eldoutilities.serialization.SerializationUtil;
+import de.eldoria.eldoutilities.simplecommands.TabCompleteUtil;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class DatabasePresetRegistry implements PresetRegistry {
+
+    private static final Gson GSON = new Gson();
+    private Map<UUID, PresetContainer> playerPresets = new HashMap<>();
+    private PresetContainer globalPresets = new PresetContainerImpl();
+    private final DataSource dataSource;
+    private final Plugin plugin;
+
+    public DatabasePresetRegistry(DataSource dataSource, Plugin plugin) {
+        this.dataSource = dataSource;
+        this.plugin = plugin;
+        this.reload();
+    }
+
+    private void reload() {
+        this.playerPresets.clear();
+        this.globalPresets.getPresets().clear();
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection()) {
+                try(PreparedStatement preparedStatement = connection.prepareStatement("CREATE TABLE IF NOT EXISTS player_presets (PLAYER_UUID longtext, PLAYER_PRESET longtext)")) {
```

Comment:
You naming scheme seems to be a bit inverted. Column names are usually lower_case while the types should be UPPER_CASE.
See: https://rainbowdashlabs.github.io/BasicSQLPlugin/02/naming/

A UUID can be put in a TINYTEXT or a VARCHAR
https://rainbowdashlabs.github.io/BasicSQLPlugin/02/sql_datatypes/

The LONGTEXT type can actually hold up to 5 billion characters. that will always be enough, but a text (65k) or a mediumtext (16 million) should be suffice as well.

There is also probably the name of the preset missing, which should be at least its own column